Source/DafnyCore/AST/TopLevelDeclarations.cs

```diff
@@ -1382,7 +1385,7 @@ public class CoDatatypeDecl : DatatypeDecl {
   }
 
   public override DatatypeCtor GetGroundingCtor() {
-    return Ctors[0];
+    return Ctors.FirstOrDefault(ctor => ctor.IsGhost, Ctors[0]);
```

Comment:
The "grounding constructor" is what the compiler will use as a default value. In an inductive datatype, there are various limitations, since the grounding constructor must be a constructor whose arguments can be constructed. But for a codatatype, any any constructor will do, since any recursive calls will be hidden inside a thunk anyway. So, the code here used to pick `Ctors[0]`, rather arbitrarily.

If there is a ghost constructors, that's a great choice, because then the compiler can (essentially) leave the value uninitialized. So, that's why the code picks the first ghost constructor, if there is one, and otherwise picks the first non-ghost constructor.